components/HomeLiveBanner/index.tsx

```diff
@@ -1,16 +1,127 @@
-enum HomeLiveBannerStatus {
+import useSWR from "swr";
+import { channel } from "../../pages/watch/live/[liveURLName]";
+import {
+  AnimatePresence,
+  domAnimation,
+  LazyMotion,
+  m,
+  MotionStyle,
+  Transition,
+} from "framer-motion";
+import React, { ReactNode } from "react";
+import styles from "./index.module.css";
+import dynamic from "next/dynamic";
+
+export enum HomeLiveBannerStatus {
   null,
   live,
   scheduled,
   cancelled,
   finished,
-  featured,
 }
 
-interface HomeLiveBannerProps {
-  status: HomeLiveBannerStatus;
-}
+export default function HomeLiveBanner() {
+  const fetcher = (input: RequestInfo, init?: RequestInit) =>
+    fetch(input, init).then((res) => res.json());
+  const { data } = useSWR<Array<channel>>(
+    `${process.env.NEXT_PUBLIC_REST_API}/v1/public/playout/channel`,
+    fetcher,
+    { refreshInterval: 60000 }
+  );
+
+  const variants = {
+    initial: { height: 0 },
+    show: { height: "auto" },
+  };
+
+  const SectionWrapper = ({
+    key,
+    children,
+    style,
+    transition,
+  }: {
+    key: string;
+    children: ReactNode;
+    style?: MotionStyle;
+    transition?: Transition;
+  }) => (
+    <m.div
+      style={{ overflow: "hidden", ...style }}
+      initial={"initial"}
+      variants={variants}
+      animate={"show"}
+      exit={"initial"}
+      transition={transition ?? { duration: 0.8 }}
+      key={key}
+    >
+      {children}
+    </m.div>
+  );
+
+  const LiveModal = dynamic(() => import("../LiveFeaturedPlayerBanner"));
+
+  return (
+    <LazyMotion features={domAnimation} strict>
+      <AnimatePresence exitBeforeEnter initial={false}>
+        {data &&
+          data.length > 0 &&
+          (() => {
+            const channel = data[0];
+            let status = HomeLiveBannerStatus[channel.status];
 
-export default function HomeLiveBanner({ status }: HomeLiveBannerProps) {
-  return <div>{JSON.stringify(HomeLiveBannerStatus[status])}</div>;
+            if (channel && status == HomeLiveBannerStatus.live)
```

Comment:
Do we need to check if `channel` is truthy here? Considering we've already checked the array sized and used a property on channel already

Reply:
nope, this was a relic before a cursory refactor, fixed by d44d1105bb7d9496cd6822b5be73e269d0e6e227